### PR TITLE
get_result_entities() Doesn't support revisions

### LINKED ISSUE #3720: crossport views commits from 7.x-3.12
- [x] https://github.com/backdrop/backdrop/pull/2674 | [d55db813](http://drupalcode.org/project/views/commit/d55db813e033ea421ac41cbdf7cb76aff3915dde) | [Issue #2480753 by dawehner, jibran, das-peter: Make it easier to subclass views_handler_area_view](http://drupal.org/node/2480753)
- [x] https://github.com/backdrop/backdrop/pull/2673 | [4317c5eb](http://drupalcode.org/project/views/commit/4317c5eb4ec719c05da4185307c8ec271304508f) | [Issue #1831162 by dcmouyard, Valentine94: Tables generated by Views need better semantics](http://drupal.org/node/1831162)
- [x] https://github.com/backdrop/backdrop/pull/2672 | [d08465de](http://drupalcode.org/project/views/commit/d08465de955b88e5f262c06d99b79ceb167dca88) | [Issue #1976836 by Dean Reilly: Views locale edit link field uses the old D6 path](http://drupal.org/node/1976836)
- [x] https://github.com/backdrop/backdrop/pull/2671 | [4218722c](http://drupalcode.org/project/views/commit/4218722c7b0b61256489dd80a0c7ba1d571696b1) | [Issue #1959594 by chrisjlee: typo 'lenght' in view_trim_views()](http://drupal.org/node/1959594)
- [x] https://github.com/backdrop/backdrop/pull/2670 | [39abeb04](http://drupalcode.org/project/views/commit/39abeb04ed654553b157115f4e3176ce24ec34b6) | [Issue #2473959 by fizk: Typo](http://drupal.org/node/2473959)
- [x] https://github.com/backdrop/backdrop/pull/2669 | [2a5df673](http://drupalcode.org/project/views/commit/2a5df67307cd603fc0f98422f0454f0776468ea8) | [Issue #1819538 by azinck: More link disappears when time-based views cache is enabled](http://drupal.org/node/1819538)
- [x] N/A | [c3536f4d](http://drupalcode.org/project/views/commit/c3536f4db9762dfa84a03aa558e0027acd6620d4) | [Issue #1036962 by acbramley, dsnopek, grahamC, mr.york, das-peter, wiifm: Edit link destination incorrect when using AJAX-enabled views](http://drupal.org/node/1036962) - Fixed in Backdrop in eff5173481b797f30c2a0a2ee01d1a46dafabb10
- [x] https://github.com/backdrop/backdrop/pull/2668 | [c125e4fc](http://drupalcode.org/project/views/commit/c125e4fc95e7399b130eab54eb12eaac3172b836) | [Issue #2408939 by shashikant_chauhan: Views Plugins documentation topic lists incorrect number of plugin types](http://drupal.org/node/2408939)
- [x] https://github.com/backdrop/backdrop/pull/2667 | [d7837bc1](http://drupalcode.org/project/views/commit/d7837bc1adcef409523b7944a81b784ebefb6fa1) | [Issue #2284593 by jhedstrom, Lendude:  Fatal error: Call to a member function ensure_my_table() on a non-object](http://drupal.org/node/2284593)
- [x] N/A | [96a54e04](http://drupalcode.org/project/views/commit/96a54e041b171eb47c6a00805b9c4731c81d9283) | [Issue #1809958 by joelpittet, steve.stotter, alesr, klaasvw: Views with exposed filter (ajax enabled) inside modal window (ctools)](http://drupal.org/node/1809958)
- [x] https://github.com/backdrop/backdrop/pull/2666 | [ae60042d](http://drupalcode.org/project/views/commit/ae60042d359c1bd40cb25bd16b6dc26dbb8ae960) | [Issue #2188165 by dww, jesss, tim.plunkett: Issue tag 'is/not empty' and 'not one of' operators do not work](http://drupal.org/node/2188165)
- [x] https://github.com/backdrop/backdrop/pull/2665 | [ea75e556](http://drupalcode.org/project/views/commit/ea75e556c2c06952c90c2c03c16c5a4570c76e33) | [Issue #1484208 by tedfordgif: Undefined index options_form()](http://drupal.org/node/1484208)
- [x] https://github.com/backdrop/backdrop/pull/2664 | [28d92baa](http://drupalcode.org/project/views/commit/28d92baa758adce7329cdbd5a5b178c1d17aac22) | [Issue #2211253 by sandipmkhairnar: Incorrect array index given in example for hook_views_ajax_data_alter()](http://drupal.org/node/2211253)
- [ ] https://github.com/backdrop/backdrop/pull/2661 | [05fcf52d](http://drupalcode.org/project/views/commit/05fcf52dd3b374513320a3e6b7c67757baf6495d) | [Issue #1802438 by andrewbelcher, vasike: get_result_entities() Doesn't support revisions](http://drupal.org/node/1802438)
- [x] https://github.com/backdrop/backdrop/pull/2660 | [21fd8f9f](http://drupalcode.org/project/views/commit/21fd8f9f89deaf875e768b6fb97717297f0f6967) | [Issue #2115269 by lalweil, anrikun: Check for entity in views_handler_field_field->get_value() before accessing it.](http://drupal.org/node/2115269)
- [x] https://github.com/backdrop/backdrop/pull/2659 | [24188c4f](http://drupalcode.org/project/views/commit/24188c4f20181a113f3b86219971f70169c94c94) | [Issue #339384 by colan: Added help text to keep the exposed filter terms list unset to keep unfiltered "Any" option default.](http://drupal.org/node/339384)
- [x] https://github.com/backdrop/backdrop/pull/2658 | [6d8b0cb5](http://drupalcode.org/project/views/commit/6d8b0cb57927569c709b77fd2ff1df0c809f2327) | [Issue #2376607 by rrfegade, dakku: Fix some typos in comments.](http://drupal.org/node/2376607)
- [x] https://github.com/backdrop/backdrop/pull/2657 | [0f54f7fd](http://drupalcode.org/project/views/commit/0f54f7fd228881357e24927fa30d46cd8e96fd94) | [Issue #1491082 by drzraf, dawehner, colan: Set a default template if there is no theme definition.](http://drupal.org/node/1491082)
- [x] https://github.com/backdrop/backdrop/pull/2656 | [4ae33535](http://drupalcode.org/project/views/commit/4ae3353503cea91b9e4e7106a7c9eca63bc4064a) | [Issue #1771140 by imclean, das-peter, GuGuss: view_menu_alter() adding to existing "access arguments" causes user_access() error](http://drupal.org/node/1771140)
- [x] https://github.com/backdrop/backdrop/pull/2655 | [d66cd3e6](http://drupalcode.org/project/views/commit/d66cd3e6a75fbab75548106ea666ddd7b6293468) | [Issue #1547924 by rbp: Document field_math feature previous value](http://drupal.org/node/1547924)
- [x] N/A See https://github.com/backdrop/backdrop-issues/issues/4137 | [fba32b8e](http://drupalcode.org/project/views/commit/fba32b8efcff01e5fd03a0905aa1647d6dfea52b) | [Issue #1248716 by milesw, ndf, stefan.r: Views strings not translatable when loaded from a feature](http://drupal.org/node/1248716)
- [x] https://github.com/backdrop/backdrop/pull/2653 | [5a0309a2](http://drupalcode.org/project/views/commit/5a0309a25040dba327613ed64b58570bb1f25f7c) | [Issue #2559839 by hexabinaer: [Backport] Expand display "Comment" trim from 10 to 80 characters](http://drupal.org/node/2559839)
- [x] https://github.com/backdrop/backdrop/pull/2652 | [f510a924](http://drupalcode.org/project/views/commit/f510a924eb8c7da4c105dee2ad381a58d86ed9b6) | [Issue #1940212 by jweowu: "Content translation: Translations" relationship incorrectly specifying translation source](http://drupal.org/node/1940212)
- [x] N/A | [8e0dab36](http://drupalcode.org/project/views/commit/8e0dab36d94a287e7fa79b61568d7d495adb10b5) | [Issue #2579819 by skyredwang: Deprecating PHP4 style constructors](http://drupal.org/node/2579819)
- [x] N/A | [dd39988f](http://drupalcode.org/project/views/commit/dd39988f1dfd4876b5a30d1e6c5f5b2e373f04dc) | [Issue #2450447: Drupal QA is broken for Views 7.x-3.x](http://drupal.org/node/2450447) - Already fixed in Backdrop per eff5173481b797f30c2a0a2ee01d1a46dafabb10.
- [x] N/A (see #3775) | [d649d329](http://drupalcode.org/project/views/commit/d649d329eb2fc56d9feb1320c173fd69f3c9a986) | [Issue #2204257 by ezra-g, andyg5000: Update Views Content access filter per core performance improvements](http://drupal.org/node/2204257)  https://github.com/backdrop/backdrop/pull/2651
- [x] https://github.com/backdrop/backdrop/pull/2679 [see comment/question](https://github.com/backdrop/backdrop-issues/issues/3720#issuecomment-491753374) | [4edefe17](http://drupalcode.org/project/views/commit/4edefe170aec2df46a59a3b5a17e67429f9108fb) | [Issue #1154830 by heddn, hass, dunix: Schema mismatch after upgrade D6 to D7](http://drupal.org/node/1154830)
- [x] https://github.com/backdrop/backdrop/pull/2650 | [9aaf9692](http://drupalcode.org/project/views/commit/9aaf9692081ba5ca3a79137069a0d508f78a9cb4) | [Issue #1337894 by fengtan: Views RSS feeds have validation issues](http://drupal.org/node/1337894)
- [x] https://github.com/backdrop/backdrop/pull/2649 | [0ae5fd66](http://drupalcode.org/project/views/commit/0ae5fd6650e52fa1f32d385ab8adfaa15abb6966) | [Issue #1751424 by Ken Ficara, jwilson3: Undefined index: current_user](http://drupal.org/node/1751424)

### PR DESCRIPTION
**Pull request by klonos**
https://www.drupal.org/node/1802438

https://git.drupalcode.org/project/views/commit/05fcf52dd3b374513320a3e6b7c67757baf6495d

https://github.com/backdrop/backdrop-issues/issues/3720

Fixes https://github.com/backdrop/backdrop-issues/issues/3720

### CODE REVIEW

File: core/modules/views/plugins/views_plugin_query_default.inc [hunk 1]
```diff
@@ -1624,12 +1625,44 @@ class views_plugin_query_default extends views_plugin_query {
...
+    else {
+      // There's no way in core to load revisions in bulk.
+      $result = array();
+      foreach ($ids as $key => $id) {
+        // Nodes can be dealt with in core.
+        if ($entity_type == 'node') {
+          $result[$key] = node_load(NULL, $id);
+        }
+        // Otherwise, see if entity is enabled.
+        elseif (module_exists('entity')) {
```
Comment (docwilmot) [inline core/modules/views/plugins/views_plugin_query_default.inc]: Entity is a required module in Backdrop, it should always exist, no?

Comment (quicksketch) [reply core/modules/views/plugins/views_plugin_query_default.inc]: Yeah agreed, we can remove the node-specific condition and just work with the entity_load_multiple() approach for everything including nodes.

**Comment by herbdool** [comment]
I've updated with the suggestions.

**Comment by quicksketch** [comment]
Thanks @herbdool! This looks very different from the Drupal fix, considering our Entity system is quite a bit different than D7. The original Drupal issue seemed to only show itself when using a contrib module. Can you provide steps to demonstrate the bug before and after this fix?

**Comment by herbdool** [comment]
Good question! I haven't tested for a bug before and after, so I won't be able to provide details. I just cleaned up @klonos PR a bit. I say we take this off the milestone and leave it until someone does provide testing steps.

File: core/modules/views/plugins/views_plugin_query_default.inc [hunk 2]
```diff
@@ -1614,7 +1614,8 @@ class views_plugin_query_default extends views_plugin_query {
     }
     $entity_type = $table_data['table']['entity type'];
     $info = entity_get_info($entity_type);
-    $id_alias = $this->get_field_alias($base_table_alias, $info['entity keys']['id']);
+    $is_revision = !empty($table_data['table']['revision']);
+    $id_alias = $this->get_field_alias($base_table_alias, $info['entity keys'][$is_revision ? 'revision' : 'id']);
```
Comment (jenlampton) [inline core/modules/views/plugins/views_plugin_query_default.inc:1618]: This is hard to read, can we split it into:
```php
if (!$is_revision) {
  $id_alias = $this->get_field_alias($base_table_alias, $info['entity keys']['id']);
}
else {
  $id_alias = $this->get_field_alias($base_table_alias, $info['entity keys']['revision']);
}
```

**Comment by quicksketch** [comment]
I gave another review and at least tested that out-of-box node listings work as expected. The code still looks fine to me too. Merged into 1.x and 1.26.x.

**Pull request merged**